### PR TITLE
Task/upgrade yarn

### LINKED ISSUE #1296: Upgrade Yarn to version 2
**Describe the task**
Upgrade to yarn v2
- https://yarnpkg.com/getting-started/migration#step-by-step
Follow steps and commit changed files

**Acceptance Criteria**
- [ ] `main` successfully builds and deploys our FE using yarnv2

**Additional context**
- Add any other context about the task here.
- Or here

### PR DESCRIPTION
**Pull request by jmetev1**
1. upgrades yarn. 
 i've been working on this for a while so not positive what will be required when others work in web/. 
Running yarn install might work. or may have to first do `corepack enable` `yarn set version berry`
current npmrc might work. or might have to do 
```
 yarn config set npmScopes.psu.npmRegistryServer https://artifacts.developer.gov.bc.ca/artifactory/api/npm/pe1e-psu-npm-local/
 yarn config set npmScopes.psu.npmAlwaysAuth true 
 yarn config set npmScopes.psu.npmAuthToken ${ARTIFACTORY_SVCACCT_TOKEN}
```
2. reorders dockerfile.web steps for better caching
# Test Links:
[Landing Page](https://wps-pr-3791-e1e498-dev.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-3791-e1e498-dev.apps.silver.devops.gov.bc.ca/morecast)
[Percentile Calculator](https://wps-pr-3791-e1e498-dev.apps.silver.devops.gov.bc.ca/percentile-calculator)
[C-Haines](https://wps-pr-3791-e1e498-dev.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-3791-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-3791-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-3791-e1e498-dev.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-3791-e1e498-dev.apps.silver.devops.gov.bc.ca/hfi-calculator)
closes #1296

### CODE REVIEW

File: Dockerfile.web [hunk 1]
```diff
@@ -14,12 +14,18 @@ FROM registry.access.redhat.com/ubi9/nodejs-20 as static
...
+COPY web/yarn.lock . 
+COPY web/.yarnrc.yml .
+# do install first so it will be cached
+RUN npm install -g --ignore-scripts corepack
+RUN corepack enable
+RUN yarn set version berry
+RUN yarn config set npmScopes.psu.npmRegistryServer https://artifacts.developer.gov.bc.ca/artifactory/api/npm/pe1e-psu-npm-local/
+RUN yarn config set npmScopes.psu.npmAlwaysAuth true 
+RUN yarn config set npmScopes.psu.npmAuthToken ${ARTIFACTORY_SVCACCT_TOKEN}
+RUN CYPRESS_INSTALL_BINARY=0 yarn install --immutable
```
Comment (jmetev1) [inline Dockerfile.web:27]: immutable is new "frozen lockfile"

File: web/package.json [hunk 2]
```diff
@@ -47,6 +47,7 @@
     "lodash": "^4.17.21",
     "luxon": "^3.0.2",
     "match-sorter": "^6.3.1",
+    "nyc": "^17.0.0",
```
Comment (jmetev1) [inline web/package.json:50]: something about yarn and bin files means this needs to be installed at top level now for merge coverage script to work.

File: Dockerfile.web [hunk 3]
```diff
@@ -5,7 +5,7 @@
 # the cpu pool size to something realistic.
 ARG NODE_OPTIONS="--v8-pool-size=4"
 
-# PHASE 1 - build static html.
+# PHASE 1 - build frontend.
 # Pull from local registry - we can't pull from docker due to limits.
 # see https://catalog.redhat.com/software/containers/ubi8/nodejs-14/5ed7887dd70cc50e69c2fabb for
```
Comment (dgboss) [inline Dockerfile.web:10]:
```suggestion
# see https://catalog.redhat.com/software/containers/ubi9/nodejs-20/64770ac7a835530172eee6a9 for
```

Comment (dgboss) [reply Dockerfile.web:10]: Sorry, missed this stale comment in the previous PR. Would you mind slipping in this update?

File: .github/workflows/post_merge_integration.yml [hunk 4]
```diff
@@ -207,6 +207,10 @@ jobs:
         run: npm config set @psu:registry https://artifacts.developer.gov.bc.ca/artifactory/api/npm/pe1e-psu-npm-local/ && npm config set //artifacts.developer.gov.bc.ca/artifactory/api/npm/pe1e-psu-npm-local/:_authToken ${{ secrets.ARTIFACTORY_SVCACCT_TOKEN }}
       - name: Install node dependencies (web)
         working-directory: ./web
+      - name: enable corepack
+        run: corepack enable
+      - name: use new yarn
+        run: yarn set version berry
```
Comment (dgboss) [inline .github/workflows/post_merge_integration.yml:213]: Do we need to setup yarn for scoped artifactory packages here like was done in `integration.yml`?

Comment (jmetev1) [reply .github/workflows/post_merge_integration.yml:210]: ah good catch

Comment (jmetev1) [reply .github/workflows/post_merge_integration.yml:210]: fixed

**Review by conbrad — APPROVED** [review]

**Review by dgboss — APPROVED** [review]
Looks great! Thank you!

**Review by brettedw — APPROVED** [review]

**Pull request merged**